packages/account/src/providers/provider.ts

```diff
@@ -814,10 +811,11 @@ Supported fuel-core version: ${supportedVersion}.`
    *
    * @returns the base asset ID.
    */
-  getBaseAssetId() {
+  async getBaseAssetId() {
```

Comment:
Wouldn't changing these methods to be async make this PR a breaking change?

Reply:
Good catch! You're right, unfortunately.

I'll take a second look and either go with the breaking changes or find a middle ground.

Reply:
I'm incorporating the breaking changes and will mark the PR as ready when I'm done.